osu.Game/Graphics/Cursor/CursorTrail.cs

```diff
@@ -0,0 +1,155 @@
+﻿//Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
+//Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Batches;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shaders;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Textures;
+using osu.Framework.Input;
+using osu.Framework.MathUtils;
+using OpenTK;
+
+namespace osu.Game.Graphics.Cursor
+{
+    class TrailSprite : Sprite
+    {
+        private Shader shader;
+
+        public TrailSprite()
+        {
+            Origin = Anchor.Centre;
+        }
+
+        protected override void ApplyDrawNode(DrawNode node)
+        {
+            base.ApplyDrawNode(node);
+
+            SpriteDrawNode sNode = node as SpriteDrawNode;
+            sNode.RoundedTextureShader = sNode.TextureShader = shader;

```

Comment:
Using `as` without null check(or null propagation) is a bit weird. I think it should be just `(SpriteDrawNode) node`.